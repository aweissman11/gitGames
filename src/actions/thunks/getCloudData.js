import { fetchCall } from '../../utilities/fetchCall';

import { loadingWordCloud, hasErrored, setCloudData } from '../'

import { mockWordCloudData } from './__mocks__/mockWordCloudData';

export const getCloudData = (user) => {
  return async (dispatch) => {
    let wordCloudData;
    dispatch(loadingWordCloud('Getting Cloud Data', true))

    try {
      const cloudUrl = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/commit_messages?username=${user.username}`
      wordCloudData = await fetchCall(cloudUrl)
    } catch(error) {
      dispatch(hasErrored('Cloud data fetch failed', true))
      return;
    }

    dispatch(setCloudData(wordCloudData))
    dispatch(loadingWordCloud('Cloud data retrieved', false))
  }
}


// import { fetchCall } from '../../utilities/fetchCall';

// import { loadingWordCloud, hasErrored, setCloudData } from '../'

// import { mockWordCloudData } from './__mocks__/mockWordCloudData';

// export const getCloudData = (user) => {
//   return async (dispatch) => {
//     let wordCloudData;
//     dispatch(loadingWordCloud('Getting Cloud Data', true))
//     dispatch(setCloudData(mockWordCloudData))
//     dispatch(loadingWordCloud('Cloud data retrieved', false))
//   }
// }