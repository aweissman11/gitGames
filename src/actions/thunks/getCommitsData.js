import { fetchCall } from '../../utilities/fetchCall';

import { loadingCommits, hasErrored, setCommitsData } from '../'

import { mockCommitsData } from './__mocks__/mockCommitsData';

export const getCommitsData = (user) => {
  return async (dispatch) => {
    let commitsData;
    dispatch(loadingCommits('Getting Commits Data', true))

    try {
      const cloudUrl = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/commit_timelines?username=${user.username}`
      commitsData = await fetchCall(cloudUrl)
    } catch(error) {
      dispatch(hasErrored('Commits data fetch failed', true))
      return;
    }

    dispatch(setCommitsData(commitsData))
    dispatch(loadingCommits('Commits data retrieved', false))
  }
}


// import { fetchCall } from '../../utilities/fetchCall';

// import { loadingCommits, hasErrored, setCommitsData } from '../'

// import { mockCommitsData } from './__mocks__/mockCommitsData';

// export const getCommitsData = (user) => {
//   return async (dispatch) => {
//     let commitsData;
//     dispatch(loadingCommits('Getting Commits Data', true))
//     dispatch(setCommitsData(mockCommitsData))
//     dispatch(loadingCommits('Commits data retrieved', false))
//   }
// }