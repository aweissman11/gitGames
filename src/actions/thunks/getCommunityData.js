import { fetchCall } from '../../utilities/fetchCall';

import { loadingCommunity, hasErrored, setCommunityData } from '../'

// import { mockCommunityData } from './__mocks__/mockCommunityData';

export const getCommunityData = (user) => {
  return async (dispatch) => {
    let communityData;
    dispatch(loadingCommunity('Pulling info on git community involvment', true))

    try {
      const communityUrl = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/community?username=${user.username}`
      communityData = await fetchCall(communityUrl)
    } catch(error) {
      dispatch(hasErrored('Community data fetch failed', true))
      return;
    }

    dispatch(setCommunityData(communityData))
    dispatch(loadingCommunity('Community data retrieved', false))
  }
}


// import { fetchCall } from '../../utilities/fetchCall';

// import { loadingCommunity, hasErrored, setCommunityData } from '../'

// import { mockCommunityData } from './__mocks__/mockCommunityData';

// export const getCommunityData = (user) => {
//   return async (dispatch) => {
//     let communityData;
//     dispatch(loadingCommunity('Getting Community Data', true))
//     dispatch(setCommunityData(mockCommunityData))
//     dispatch(loadingCommunity('Community data retrieved', false))
//   }
// }