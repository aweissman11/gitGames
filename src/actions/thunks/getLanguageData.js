import { fetchCall } from '../../utilities/fetchCall';

import { loadingLanguages, hasErrored, setLanguageData } from '../'

export const getLanguageData = (user) => {
  return async (dispatch) => {
    let languagesData;
    dispatch(loadingLanguages('Getting Languages Data', true))

    try {
      const cloudUrl = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/languages?username=${user.username}`
      languagesData = await fetchCall(cloudUrl)
    } catch(error) {
      dispatch(hasErrored('Languages data fetch failed', true))
      return;
    }
    dispatch(setLanguageData(languagesData))
    dispatch(loadingLanguages('Languages data retrieved', false))
  }
}