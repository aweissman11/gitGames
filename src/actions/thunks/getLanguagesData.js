// import { fetchCall } from '../../utilities/fetchCall';

// import { loadingLanguages, hasErrored, setLanguageData } from '..';

// import { mockLanguagesData } from './__mocks__/mockLanguagesData';

// export const getLanguageData = (user) => {
//   return async (dispatch) => {
//     let languagesData;
//     dispatch(loadingLanguages('Getting Languages Data', true))

//     try {
//       const cloudUrl = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/languages?username=${user.username}`
//       languagesData = await fetchCall(cloudUrl)
//     } catch(error) {
//       dispatch(hasErrored('Languages data fetch failed', true))
//       return;
//     }

//     dispatch(setLanguageData(languagesData))
//     dispatch(loadingLanguages('Languages data retrieved', false))
//   }
// }


import { fetchCall } from '../../utilities/fetchCall';

import { loadingLanguages, hasErrored, setLanguageData } from '..';

import { mockLanguagesData } from './__mocks__/mockLanguagesData';

export const getLanguageData = (user) => {
  return async (dispatch) => {
    let languagesData;
    dispatch(loadingLanguages('Getting Languages Data', true))
    dispatch(setLanguageData(mockLanguagesData))
    dispatch(loadingLanguages('Languages data retrieved', false))
  }
}