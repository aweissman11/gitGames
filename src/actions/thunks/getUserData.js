import { fetchCall } from '../../utilities/fetchCall';

import { isLoading, hasErrored, setUserData } from '../'
import { getCloudData } from './getCloudData';
import { getLanguageData } from './getLanguageData';

export const getUserData = (user) => {
  return async (dispatch) => {
    let userData;
    let wordCloudData;
    dispatch(isLoading('Getting User Data', true))
    try {
      const userUrl = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/users?username=${user.username}`
      userData = await fetchCall(userUrl)
    } catch(error) {
      dispatch(hasErrored('User data fetch failed', true))
      return;
    }

    dispatch(getCloudData(user))
    dispatch(getLanguageData(user))
    dispatch(setUserData(userData));
    dispatch(isLoading('User data retrieved', false))
  }
}