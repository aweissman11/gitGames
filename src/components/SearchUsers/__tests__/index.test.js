import React from 'react';
import { shallow } from 'enzyme';

import { SearchUsers } from '../index';
import { getUserData } from '../../../actions/thunks/getUserData';

import { mapDispatchToProps } from '../index';

describe('SearchUsers', () => {
  
  let wrapper;
  let mockEvent;

  const mockPreventDefault = jest.fn();
  const mockGetUserData = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SearchUsers 
                        loginPage={{user: 'me'}} 
                        getUserData={mockGetUserData}
                        />)
    mockEvent = {
      target: {
        value: 'ev',
      },
      preventDefault: mockPreventDefault
    }
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should MDTP', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.getUserData();


    expect(mockDispatch).toHaveBeenCalled()
  })

  it('handleSearchChange should set state', () => {
    wrapper.instance().handleSearchChange(mockEvent);
    expect(wrapper.instance().state.query).toEqual('ev')
  })
  
  it('should call handleSearchChange on text input', () => {
    const mockHandleChange = jest.fn();
    wrapper.instance().handleSearchChange = mockHandleChange;
    wrapper.instance().toggleSearchInput();
    wrapper.find('#search-input').simulate('change', mockEvent);
    expect(mockHandleChange).toHaveBeenCalledWith(mockEvent);
  });

  describe('toggleSearchInput', () => {

    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<SearchUsers loginPage={{user: 'me'}} />)
    })

    it('should set to the right state', () => {
      wrapper.instance().toggleSearchInput();
      expect(wrapper.instance().state).toEqual({
        "boxWidth": "search-box-width",
        "formWidth": "search-form-width",
        "inputWidth": "search-input-width",
        "query": "",
        "searchInputIsToggled": true,
        "searchUsers": false,
        "svgToggled": "toggled"
      })
    })

    it('should set to the right state if search is toggled already', () => {
      wrapper.instance().toggleSearchInput();
      wrapper.instance().toggleSearchInput();
      expect(wrapper.instance().state).toEqual({
        "boxWidth": "",
        "formWidth": "",
        "inputWidth": "",
        "query": "",
        "searchInputIsToggled": false,
        "searchUsers": false,
        "svgToggled": ""
      })
    })
  })

  describe('submitSearch', () => {
    it('should call preventDefault', () => {
      wrapper.instance().submitSearch(mockEvent);
      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it('should call getUserData with the correct parameters', () => {
      wrapper.instance().toggleSearchInput();
      wrapper.instance().submitSearch(mockEvent);

      expect(mockGetUserData).toHaveBeenCalledWith({ username: ''})
    });
  });
})
