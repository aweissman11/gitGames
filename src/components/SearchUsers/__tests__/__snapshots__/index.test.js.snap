// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchUsers should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchUsers
    getUserData={[MockFunction]}
    loginPage={
      Object {
        "user": "me",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="search-form "
          onSubmit={[Function]}
        >
          <input
            className="search-input "
            id="search-input"
            onChange={[Function]}
            placeholder="Search GitHub Users..."
            value=""
          />
        </form>,
        <svg
          className="svg-search "
          height="60px"
          onClick={[Function]}
          viewBox="0 0 30 30"
          width="60px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 13 3 C 7.4889971 3 3 7.4889971 3 13 C 3 18.511003 7.4889971 23 13 23 C 15.396508 23 17.597385 22.148986 19.322266 20.736328 L 25.292969 26.707031 A 1.0001 1.0001 0 1 0 26.707031 25.292969 L 20.736328 19.322266 C 22.148986 17.597385 23 15.396508 23 13 C 23 7.4889971 18.511003 3 13 3 z M 13 5 C 17.430123 5 21 8.5698774 21 13 C 21 17.430123 17.430123 21 13 21 C 8.5698774 21 5 17.430123 5 13 C 5 8.5698774 8.5698774 5 13 5 z"
          />
        </svg>,
      ],
      "className": "search-box ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="search-input "
            id="search-input"
            onChange={[Function]}
            placeholder="Search GitHub Users..."
            value=""
          />,
          "className": "search-form ",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input ",
            "id": "search-input",
            "onChange": [Function],
            "placeholder": "Search GitHub Users...",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <path
            d="M 13 3 C 7.4889971 3 3 7.4889971 3 13 C 3 18.511003 7.4889971 23 13 23 C 15.396508 23 17.597385 22.148986 19.322266 20.736328 L 25.292969 26.707031 A 1.0001 1.0001 0 1 0 26.707031 25.292969 L 20.736328 19.322266 C 22.148986 17.597385 23 15.396508 23 13 C 23 7.4889971 18.511003 3 13 3 z M 13 5 C 17.430123 5 21 8.5698774 21 13 C 21 17.430123 17.430123 21 13 21 C 8.5698774 21 5 17.430123 5 13 C 5 8.5698774 8.5698774 5 13 5 z"
          />,
          "className": "svg-search ",
          "height": "60px",
          "onClick": [Function],
          "viewBox": "0 0 30 30",
          "width": "60px",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M 13 3 C 7.4889971 3 3 7.4889971 3 13 C 3 18.511003 7.4889971 23 13 23 C 15.396508 23 17.597385 22.148986 19.322266 20.736328 L 25.292969 26.707031 A 1.0001 1.0001 0 1 0 26.707031 25.292969 L 20.736328 19.322266 C 22.148986 17.597385 23 15.396508 23 13 C 23 7.4889971 18.511003 3 13 3 z M 13 5 C 17.430123 5 21 8.5698774 21 13 C 21 17.430123 17.430123 21 13 21 C 8.5698774 21 5 17.430123 5 13 C 5 8.5698774 8.5698774 5 13 5 z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        "type": "svg",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="search-form "
            onSubmit={[Function]}
          >
            <input
              className="search-input "
              id="search-input"
              onChange={[Function]}
              placeholder="Search GitHub Users..."
              value=""
            />
          </form>,
          <svg
            className="svg-search "
            height="60px"
            onClick={[Function]}
            viewBox="0 0 30 30"
            width="60px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M 13 3 C 7.4889971 3 3 7.4889971 3 13 C 3 18.511003 7.4889971 23 13 23 C 15.396508 23 17.597385 22.148986 19.322266 20.736328 L 25.292969 26.707031 A 1.0001 1.0001 0 1 0 26.707031 25.292969 L 20.736328 19.322266 C 22.148986 17.597385 23 15.396508 23 13 C 23 7.4889971 18.511003 3 13 3 z M 13 5 C 17.430123 5 21 8.5698774 21 13 C 21 17.430123 17.430123 21 13 21 C 8.5698774 21 5 17.430123 5 13 C 5 8.5698774 8.5698774 5 13 5 z"
            />
          </svg>,
        ],
        "className": "search-box ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="search-input "
              id="search-input"
              onChange={[Function]}
              placeholder="Search GitHub Users..."
              value=""
            />,
            "className": "search-form ",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input ",
              "id": "search-input",
              "onChange": [Function],
              "placeholder": "Search GitHub Users...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <path
              d="M 13 3 C 7.4889971 3 3 7.4889971 3 13 C 3 18.511003 7.4889971 23 13 23 C 15.396508 23 17.597385 22.148986 19.322266 20.736328 L 25.292969 26.707031 A 1.0001 1.0001 0 1 0 26.707031 25.292969 L 20.736328 19.322266 C 22.148986 17.597385 23 15.396508 23 13 C 23 7.4889971 18.511003 3 13 3 z M 13 5 C 17.430123 5 21 8.5698774 21 13 C 21 17.430123 17.430123 21 13 21 C 8.5698774 21 5 17.430123 5 13 C 5 8.5698774 8.5698774 5 13 5 z"
            />,
            "className": "svg-search ",
            "height": "60px",
            "onClick": [Function],
            "viewBox": "0 0 30 30",
            "width": "60px",
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M 13 3 C 7.4889971 3 3 7.4889971 3 13 C 3 18.511003 7.4889971 23 13 23 C 15.396508 23 17.597385 22.148986 19.322266 20.736328 L 25.292969 26.707031 A 1.0001 1.0001 0 1 0 26.707031 25.292969 L 20.736328 19.322266 C 22.148986 17.597385 23 15.396508 23 13 C 23 7.4889971 18.511003 3 13 3 z M 13 5 C 17.430123 5 21 8.5698774 21 13 C 21 17.430123 17.430123 21 13 21 C 8.5698774 21 5 17.430123 5 13 C 5 8.5698774 8.5698774 5 13 5 z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          "type": "svg",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
